metricflow/plan_conversion/node_processor.py

```diff
@@ -59,6 +59,16 @@ class MultiHopJoinCandidate:
     lineage: MultiHopJoinCandidateLineage
 
 
+@dataclass(frozen=True)
+class PredicatePushdownParameters:
```

Comment:
Nit: The issue that I've run into with classes that are plural is that when you have a variable that is a collection of them, it's unclear how to name the variable. I've been using `...ParameterSet` instead, but up for other ideas.

Reply:
Oh yeah, good point. I don't think we'd ever want to collect these but also I never like having the name parameters in a class. It's not utils-level bad but it still basically means "bag of crap."

Maybe PredicatePushdownState? It effectively tracks the known predicates and whether or not to attempt to push them down, so at any point you have a representation of state you can use to apply a constraint (or not).

Reply:
Oops I already use PredicatePushdownState as a state enumeration later on. 🤦 

`PredicatePushdownDirective`? `PredicatePushdownEvaluationInput`? `ThingsWeNeedToDoPredicatePushdownCorrectly`?

I'm going with directive because it's least verbose. We can rename later if we need to.

Reply:
@plypaul PredicatePushdownState is now available! Renamed in https://github.com/dbt-labs/metricflow/pull/1224 - tried to do it here but it was a huge pain.